lib/ui/screens/overview_screen.dart

```diff
@@ -60,16 +69,24 @@ class OverviewScreen extends StatelessWidget {
     );
   }
 
-  Widget _buildInfoBanner(BuildContext context, UserData user) {
-    final delivery = context.watch<DeliveryNotifier>().delivery;
-    if (user is! Canteen || !HelperService.canDonate(context)) {
+  Future<Widget> _buildInfoBanner(BuildContext context, UserData user) async {
+    final deliveryNotifier = context.watch<DeliveryNotifier>();
+    final canDonate = await HelperService.canDonate(context);
+
+    if (user is! Canteen || !canDonate) {
       return const SliverToBoxAdapter(child: SizedBox());
     }
 
-    return delivery?.state == DeliveryState.accepted ||
-            delivery?.state == DeliveryState.offered
-        ? _buildDeliveryConfirmedBanner(context, user)
-        : _buildDonationCountdownBanner(context);
+    if (deliveryNotifier.delivery?.state == DeliveryState.accepted ||
```

Comment:
This is just fine, but I think that we should use switch block for this. Right now it seems to be maybe too heavy, but in the future, if we add another case, it will force us to think, whether that case should return true or false.
I may be wrong so let me know what you think

Reply:
I have edited